Code/Tools/TestImpactFramework/Runtime/Common/Code/Include/Static/Dependency/TestImpactTestSelectorAndPrioritizer.h

```diff
@@ -116,7 +140,143 @@ namespace TestImpact
+                {
+                    return SourceOperationActionResult::ConcludeSelection;
+                }
+            }
+        }
+    
+        return SourceOperationActionResult::Continue;
+    }
+    
+    template<typename ProductionTarget, typename TestTarget>
+    typename TestSelectorAndPrioritizer<ProductionTarget, TestTarget>::SourceOperationActionResult
+    TestSelectorAndPrioritizer<ProductionTarget, TestTarget>::IterateUpdateParentedSourcesWithCoverage(
+        const SourceDependency<ProductionTarget, TestTarget>& sourceDependency,
+        SelectedTestTargetAndDependerMap& selectedTestTargetMap)
+    {
+        for (const auto& parentTarget : sourceDependency.GetParentTargets())
+        {
+            if (parentTarget.GetTarget()->GetType() == TargetType::ProductionTarget)
+            {
+                // Parent Targets: Yes
+                // Coverage Data : Yes
+                // Source Type   : Production
+                //
+                // Scenario
+                // 1. The existing file has been modified
+                // 2. This file exists in one or more source to production target mapping artifacts
+                // 3. There exists coverage data for this file in the source covering test list
+                if (SourceOperationActionResult::ConcludeSelection ==
+                    UpdateProductionSourceWithCoverageAction(parentTarget.GetProductionTarget(), selectedTestTargetMap, sourceDependency))
+                {
+                    return SourceOperationActionResult::ConcludeSelection;
+                }
+            }
+            else
+            {
+                // Parent Targets: Yes
+                // Coverage Data : Yes
+                // Source Type   : Test
+                //
+                // Scenario
+                // 1. The existing file has been modified
+                // 2. This file exists in one or more source to test target mapping artifacts
+                // 3. There exists coverage data for this file in the source covering test list
+                if (SourceOperationActionResult::ConcludeSelection ==
+                    UpdateTestSourceWithCoverageAction(parentTarget.GetTestTarget(), selectedTestTargetMap))
+                {
+                    return SourceOperationActionResult::ConcludeSelection;
+                }
+            }
+        }
+    
+        return SourceOperationActionResult::Continue;
+    }
+    
+    template<typename ProductionTarget, typename TestTarget>
+    typename TestSelectorAndPrioritizer<ProductionTarget, TestTarget>::SourceOperationActionResult
+    TestSelectorAndPrioritizer<ProductionTarget, TestTarget>::IterateUpdateParentedSourcesWithoutCoverage(
+        const SourceDependency<ProductionTarget, TestTarget>& sourceDependency,
+        SelectedTestTargetAndDependerMap& selectedTestTargetMap)
+    {
```

Comment:
suggestion: Maybe some of this code should be moved to an impl file so it's not all in the header (not in this PR but maybe a follow-up?)